### PR TITLE
Post on transformation learning

### PR DESCRIPTION
**Pull request by Dref360**

### CODE REVIEW

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+author: Frédéric Branchaud-Charron
+pdf: http://papers.nips.cc/paper/7263-unsupervised-transformation-learning-via-convex-relaxations.pdf
+cite:
+  authors: "Tatsunori B., Hashimoto John C. and Duchi Percy Liang"
+  title:   "Unsupervised Transformation Learning
+via Convex Relaxations"
+  venue:   "NIPS 2017"
+---
+
+The authors propose an unsupervised method to estimate the transformation matrix to describe a dataset.
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: Suggestion: "method to discover transformation matrices associated to relationships in the dataset."

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+To do so, the method looks at the closest neighbor of some samples and optimize a novel convex problem.
+
+
+## Problem
+Given an input $$x \in \Re^d$$ we want a matrix $$A^{d \times d} | tAx + x = \bar x$$, where $$t$$ is a scaling factor and $$\bar x$$ is the nearest neigbhor.
+
+We want to minize the following equation :
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: minimize

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+![](/machine-learning/images/translearning/toopti.png)
+## Learning transformations
+
+The authors use Lie groups to define every transformation. A Lie group is a group for which the multiplication ($$\Re^d \rightarrow \Re^d$$) and the inversion are differentiable.
+
+We want to minimzed a weighted sum of vectors which would approximate the real transformation. The weights are noted as $$\alpha$$
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: minimize

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+via Convex Relaxations"
+  venue:   "NIPS 2017"
+---
+
+The authors propose an unsupervised method to estimate the transformation matrix to describe a dataset.
+To do so, the method looks at the closest neighbor of some samples and optimize a novel convex problem.
+
+
+## Problem
+Given an input $$x \in \Re^d$$ we want a matrix $$A^{d \times d} | tAx + x = \bar x$$, where $$t$$ is a scaling factor and $$\bar x$$ is the nearest neigbhor.
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: neighbor

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## Disentangling Transformations
+From a set of transformations $$A_{1..K}$$ we want to extract uncorrelated transformations. We can find those by computing the SVD of those transformations. *(See Theorem 3.4)*
+
+## Experiments
+
+It took 10 minutes to find the most importants transformations on MNIST. The authors also tried on another dataset (Kannada handwriting) without retraining. They also tested their method on CelibA (not shown here).
+
+![](/machine-learning/images/translearning/qualitative.png)
+
+Here are some quantitaive result.
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: quantitative

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+For $$K$$ transformation, we get for $$r > K^2 , A_k = \sum^{r}_{i=1}\alpha_{ik}(\bar x_{i} - x_{i})x_{i}^{T}$$
+![](/machine-learning/images/translearning/eq6.png)
+
+## Disentangling Transformations
+From a set of transformations $$A_{1..K}$$ we want to extract uncorrelated transformations. We can find those by computing the SVD of those transformations. *(See Theorem 3.4)*
+
+## Experiments
+
+It took 10 minutes to find the most importants transformations on MNIST. The authors also tried on another dataset (Kannada handwriting) without retraining. They also tested their method on CelibA (not shown here).
```
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: important

File: machine-learning/_posts/2018-01-15-translearning.md (hunk 4 above)
Comment (lemairecarl) [inline machine-learning/_posts/2018-01-15-translearning.md]: In the equation, what does the vertical bar mean? Generally it means "given", and is used in statistics. e.g. "what is the value of A, given that B = 42".
(Here, it seems like it means "such that", or "subject to".)

File: machine-learning/_posts/2018-01-15-translearning.md (hunk 6 above)
Comment (lemairecarl) [reply machine-learning/_posts/2018-01-15-translearning.md]: CelebA

File: machine-learning/_posts/2018-01-15-translearning.md (hunk 4 above)
Comment (ppoulin91) [reply machine-learning/_posts/2018-01-15-translearning.md]: It does mean "such that" when not used inside a probability :)

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+pdf: http://papers.nips.cc/paper/7263-unsupervised-transformation-learning-via-convex-relaxations.pdf
+cite:
+  authors: "Tatsunori B., Hashimoto John C. and Duchi Percy Liang"
+  title:   "Unsupervised Transformation Learning
+via Convex Relaxations"
+  venue:   "NIPS 2017"
+---
+
+The authors propose an unsupervised method to discover transformation matrices to describe a dataset.
+To do so, the method looks at the closest neighbor of some samples and optimize a novel convex problem.
```
Comment (ppoulin91) [inline machine-learning/_posts/2018-01-15-translearning.md]: "optimizes"

File: machine-learning/_posts/2018-01-15-translearning.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+## Disentangling Transformations
+From a set of transformations $$A_{1..K}$$ we want to extract uncorrelated transformations. We can find those by computing the SVD of those transformations. *(See Theorem 3.4)*
+
+## Experiments
+
+It took 10 minutes to find the most important transformations on MNIST. The authors also tried on another dataset (Kannada handwriting) without retraining. They also tested their method on CelebA (not shown here).
+
+![](/machine-learning/images/translearning/qualitative.png)
+
+Here are some quantitative result.
```
Comment (ppoulin91) [inline machine-learning/_posts/2018-01-15-translearning.md]: results

**Pull request merged**